gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -600,6 +604,12 @@ public SeekableByteChannel open(
         throw GoogleCloudStorageExceptions.getFileNotFoundException(
             resourceId.getBucketName(), resourceId.getObjectName());
       }
+
+      if (readOptions.getFastFailOnGzipEncoding()
```

Comment:
Let's move this check in GoogleCloudStorageReadChannel, it should determine during initialization if opened file is GZIP encoded

Reply:
@medb I moved check to GoogleCloudStorageReadChannel and rewrite tests, check if it is ok, please.